### ISSUE TITLE
Trans-History #1. Basic History Implementation

### ISSUE THREAD

**Issue by Louisswryan** [Neufund/platform-frontend#2821]
The final Transaction History will need to be rolled out in stages.

The first of those stages is the basic History interface shown below:

https://app.zeplin.io/project/5a8a92c89c1a166a6a6e8f37/screen/5cc6f62c6c477a3dc9b340ff

The listed events will be shown in order from most to least recent.

When a value is brought into the platform, it is represented with bold green text
When a value leaves the platform, it is represented with regular black text

For now, the included events are as follows:

In Green 

-nEUR Purchase (deposit)
-Payout/Dividend nEUR/EtherToken
-Any token transfer into platform
-Equity Token Claimed (ETO)
-ETO Refund: nEUR or Ether

In Black

-Investment nEUR/EtherToken
-nEUR Redeem
-nEUR confiscation
-Any token transfer out of platform
-ICBM Participation

The width of the History card is imprecise here buit it should meet the edges of the top cards in wallet.

If you have questions please ask : D

**Comment by rudolfix** [comment]
There's backend fixture available ask @tl-neufund for details

**Comment by rudolfix** [comment]
** tech details**
api root: `/api/analytics-api/`

getting the transactions list:
`/transactions/me`
returns a list of maximum of 100 transactions, less is possible (due to post-filtering)
with the transactions we return following fields
`version`: when present is used to refresh transaction list, it's always present when you request first page of transactions via this method
`before_transaction`: when present it means that more data is available (next page).

refreshing the transaction list:
`/transactions/new/me`
where you pass the most recent `version`. it will return **new** and **updated** transactions so you need to add or update existing transactions (can happen for example if your eurt withdrawal is settled)
returns new `version`, use it
you can poll on this method each 5 seconds, you can do it in the background when in wallet view and when you enter wallet view and some transactions are in state

getting next page:
`/transactions/me`
pass `before_transaction` you've got from previous call. transactions will not overlap with what you've got previously. 
it very rare situations you may get 0 transactions but there's still `before_transaction` present so it means that there's more data. that's because of post-filtering. should not happen really.

use fixtures:
only on `io`/`localhost` pass `use_fixtures` `true` to `/transactions/me` to get a list of all possible transaction types with several interesting states. we can add more on demand. good for e2e test

**unification of transaction types and extra data**
@desfero it seems that we have very similar schema that is used to store transactions in `user/me`. please ping me so we can
* unify transaction types (names that we use)
* unify transaction extra data - as far as it's possible

this will massively pay-off in the future

** transaction ordering and unique identifier**
transactions come ordered, the key is
```
block_number:
        type: integer
      transaction_index:
        type: integer
      log_index:
        type: integer
```
for all practical purposes `tx_hash` can be used to identify a transaction as well

**watching for pending transactions**
we should support following edge case:
1. user makes transaction on the platform
2. we bring it to success state
3. user goes to wallet and transaction is not there!

that's because it will take time for particular block height to be reached. my proposition:
* we keep hash of recently mined transaction
* when in wallet view and when transaction type is one of those supported by the backend (list is in swagger but we can expose that in api) and if `tx_hash` is not present in the transaction list, we can display an information that `your transaction list is being updated`
* if recently mined transaction is older than 60 seconds we do not show anything

**Comment by rudolfix** [comment]
**address naming**
any fields that starts ends with `_address` can have address resolved to an interface name. in that case a corresponding field `_name` will be present. for example `token_name` will have interface of the token.

you can obtain a list of interfaces from artifacts json (`KNOWN_INTERFACES`).

for each interface defined there we need a display name that we can put on the transaction view instead of address.

**data enrichment**
you can obtain a lot of interesting data both on-chain and off-chain to show to the user.
* if the address is `commitmentInterface` you can use this address to obtain `eto_data` from the backend and display whatever you want
* if the address is `equityTokenInterface` you can obtain token name and symbol + nominee and company address on-chain (see ERC20). then you can use company address to load `company_data`. you can also get commitment that was used to issue token. its in `eto_data` (only ON-CHAIN) in `equity_token_address` or alternatively you can take it on-chain via `equityTokenController`
* all tokens implement ERC-20
* please for euroToken destroy and withdraw `address_by` replace yourself with `Quintessence AG`

**testing**
* [ ] we provide fixtures for e2e that should contain all transaction types. let's talk to add more
* [ ] use `INV_HAS_EUR_HAS_KYC` - for that address we generated a lot of transaction cases

**Comment by glupiawymowka** [comment]
we will need to reconsider wording maybe. 
The action button is called "withdraw" but then the action in the history is called "transfer" which is a bit confusing

for now the Payout/return is simply marked as PAYOUT 'currency', it's not said it's from NEU. Just a note

It's also a bit confusing that we invest, let's say, 100.55 nEUR in the investment flow and no more decimals are available, but then in the history, we see 100.5500 nEUR. 
I guess that sometimes it's an equity token price (X.XXXX) and sometimes just a currency (X.XX) so this is also just a note in case someone has an idea/thought on it

**Comment by glupiawymowka** [comment]
16.07 This is not working on stage at all for now 
![image](https://user-images.githubusercontent.com/18315589/61280757-bc827c80-a7b8-11e9-98b7-4262209b5898.png)

**Comment by Louisswryan** [comment]
Hello @desfero,

Can we please make a change to this.

I think I was a bit too enthusiastic because 18px font stands out too much. Can we reduce it to the 14px standard

![Screenshot 2019-07-18 at 12.15.04.png](https://images.zenhubusercontent.com/5c5184c7df0733a30553c211/edc02b11-7d00-4ce1-9995-0f3e9adcff8e)

More in keeping with the rest of the page

![Screenshot 2019-07-18 at 12.26.08.png](https://images.zenhubusercontent.com/5c5184c7df0733a30553c211/e268356f-1557-42fc-80bb-ebaec704b90e)

Thank you

**Comment by glupiawymowka** [comment]
currently, the version on stage is tested and working (no font size changes yet, just the initial PR)

**Comment by desfero** [comment]
@Louisswryan @glupiawymowka font-size for list fixed in #2890
